src/main/scala/abalone/Pos.scala

```diff
@@ -70,10 +70,32 @@ Abalone official coordinates system and "standard start position" are drawn belo
         1 2 3 4 5
 
 
-!!! be mindful about the fact that when we write the FEN we start from topleft to bottom right
-
  */
 
+sealed trait DirectionString
+
+object DirectionString {
+  case object Left      extends DirectionString
+  case object UpLeft    extends DirectionString
+  case object UpRight   extends DirectionString
+  case object Right     extends DirectionString
+  case object DownRight extends DirectionString
+  case object DownLeft  extends DirectionString
+
+  val all: List[DirectionString] = List(Left, UpLeft, UpRight, Right, DownRight, DownLeft)
+}
+
+sealed trait DiagonalDirectionString
+
+object DiagonalDirectionString {
+  case object UpLeft    extends DiagonalDirectionString
+  case object UpRight   extends DiagonalDirectionString
+  case object DownRight extends DiagonalDirectionString
+  case object DownLeft  extends DiagonalDirectionString
+
+  val all: List[DiagonalDirectionString] = List(UpLeft, UpRight, DownRight, DownLeft)
+}
```

Comment:
it feels like four of these directions are duplicated. I spoke with copilot on how to rewrite this so that we dont define these twice, and it gave me an answer:
> In `src/main/scala/abalone/Pos.scala` in PR #195, we have two new objects, DirectionString and DiagonalDirectionString. However I note that the four case objects in DiagonalDirectionString are also duplicated in DirectionString (it seems DiagonalDirectionString.all is a subset of DirectionString.all) Is there a way to rewrite these definitions to not duplicate these four, but still have two separate 'all' lists, one with 6 elements for DirectionString and one with 4 elements for DiagonalDirectionString?